engine/system/core/src/main/java/it/unibz/inf/ontop/answering/connection/impl/QuestStatement.java

```diff
@@ -238,8 +243,10 @@ private SimpleGraphResultSet executeDescribeQuery(DescribeQuery inputQuery, Immu
 				if (describeResultSet == null) { // just for the first time
 					describeResultSet = set;
 				} else if (set != null) {
-					while (set.hasNext())
-						describeResultSet.addNewResult(set.next());
+					RDFFactCloseableIterator iterator = set.iterator();
```

Comment:
Will have to use the interface `OntopCloseableIterator<RDFFact, OntopConnectionException>`